### PR TITLE
Support array properties in bulk loader

### PR DESCRIPTION
**Pull request by jeffreylovitz**
Module-side changes to support loading array properties.

### CODE REVIEW

File: src/bulk_insert/bulk_insert.c [hunk 1]
```diff
@@ -86,6 +90,15 @@ static inline SIValue _BulkInsert_ReadProperty(const char *data, size_t *data_id
 		*data_idx += strlen(s) + 1;
 		// The string itself will be cloned when added to the GraphEntity properties.
 		v = SI_ConstStringVal((char *)s);
+	} else if(t == BI_ARRAY) {
+		// The first 8 bytes of a received array will be the array length.
+		int64_t len = *(int64_t *)&data[*data_idx];
```
Comment (swilly22) [inline src/bulk_insert/bulk_insert.c:95]: How about
```
int64_t len;
memcpy(&len, data[*data_idx], sizeof(len));
```

**Review by swilly22 — CHANGES_REQUESTED** [review]
Simply enough!
one comment + would it be possible to add a test?

File: src/bulk_insert/bulk_insert.c (hunk 1 above)
Comment (jeffreylovitz) [reply src/bulk_insert/bulk_insert.c:95]: I can change this if you like, but this is the convention used throughout this function. I can change all of them if you prefer?

**Comment by jeffreylovitz** [comment]
> Simply enough!
> one comment + would it be possible to add a test?

I've added a few tests at https://github.com/RedisGraph/redisgraph-bulk-loader/pull/37/files#diff-9d5e43080cc7b5b046d6492fc6f358f0R587-R638
I think it makes more sense to keep all the tests in the bulk loader's repo.

File: src/bulk_insert/bulk_insert.c [hunk 2]
```diff
@@ -17,7 +18,8 @@ typedef enum {
 	BI_BOOL,
 	BI_DOUBLE,
 	BI_STRING,
-	BI_LONG
+	BI_LONG,
+	BI_ARRAY,
```
Comment (swilly22) [inline src/bulk_insert/bulk_insert.c:22]: Let's explicitly number the values
```
BI_NULL = 0,
BI_BOOL = 1,
...
```

That way we can guarantee no confusion between the server and the client.

**Review by swilly22 — CHANGES_REQUESTED** [review]
Agree with your replies to my previous comments, please address this additional comment.

**Review by swilly22 — APPROVED** [review]

**Pull request merged**